clang/lib/StaticAnalyzer/Checkers/StackAddrEscapeChecker.cpp

```diff
@@ -398,7 +400,7 @@ void StackAddrEscapeChecker::checkEndFunction(const ReturnStmt *RS,
     }(Referrer->getMemorySpace());
 
     // This cast supposed to succeed.
```

Comment:
Perhaps explain why will the cast succeed (or, alternatively, just remove this meaningless comment).

Reply:
I was thinking about it once you raised this comment.
To me, to have a proper diagnostic we need something that we can name here; and they are usually variables.

Now, if we check and bail out if it's not a VarRegion, we would not get notified if we suddenly have some other value that we could cover but we don't. It took a long time to even hit this crash, so I don't think they are too frequent.

So the question is which we prefer:
 - Crash (possible in the long future) to get notified or
 - Silently ignore these unthought, interesting cases for which we didn't account for
 
 If this would be a mission-critical software I'd vote for (2), but now I'm a bit hesitant in between the two options.
 
 Maybe, given that this is a core checker, we should lean towards gracefully handling this. WDYT?

Reply:
I think the best would be either a comment that explains _why_ we expect a `VarRegion` here (e.g. it's already checked in function `fooBar()` etc.) or if possible, then a crash-less logic (converting an earlier `isa`-like check + this `cast` into a single `getAs`).

I don't think that we should add defensive/paranoid checks; if this logic is broken by some change in the distant future, then a fixable crash is better than silent degradation.

Reply:
> I think the best would be either a comment that explains _why_ we expect a `VarRegion` here (e.g. it's already checked in function `fooBar()` etc.) or if possible, then a crash-less logic (converting an earlier `isa`-like check + this `cast` into a single `getAs`).

There is nothing that would directly imply this.
It's like, ATM I don't know of any other situations where it could be not a `VarRegion`, and so far it didn't appear in practice that would disprove this.

I'm not exactly sure if a comment like that would suffice. Would you accept it? And do you think it's necessary?

Another alternative could be a combination:

```c++
if (!isa<VarRegion>(Referrer)) {
  assert(false && "We should really only have VarRegions here");
  continue;
}
// ...
cast<VarRegion>(Referrer)...
```